assets/js/modules/analytics-4/datastore/properties.test.js

```diff
@@ -0,0 +1,163 @@
+					body: { data: { accountID } },
+				} );
+
+				const properties = registry.select( STORE_NAME ).getProperties( accountID );
+				expect( properties ).toMatchObject( [ fixtures.createProperty ] );
+			} );
+
+			it( 'should dispatch an error if the request fails', async () => {
+				const accountID = fixtures.createProperty.parent;
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+
+				fetchMock.post( createPropertyEndpoint, {
+					body: response,
+					status: 500,
+				} );
+
+				await registry.dispatch( STORE_NAME ).createProperty( accountID );
+
+				const error = registry.select( STORE_NAME ).getErrorForAction( 'createProperty', [ accountID ] );
+				expect( error ).toMatchObject( response );
+
+				// The response isn't important for the test here and we intentionally don't wait for it,
+				// but the fixture is used to prevent an invariant error as the received properties
+				// taken from `response.properties` are required to be an array.
+				muteFetch( propertiesEndpoint, fixtures.properties );
+
+				const properties = registry.select( STORE_NAME ).getProperties( accountID );
+				// No properties should have been added yet, as the property creation failed.
+				expect( properties ).toBeUndefined();
+				expect( console ).toHaveErrored();
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getProperties', () => {
+			it( 'should use a resolver to make a network request', async () => {
+				fetchMock.get( propertiesEndpoint, {
+					body: fixtures.properties,
+					status: 200,
+				} );
+
+				const accountID = '12345';
+				const initialProperties = registry.select( STORE_NAME ).getProperties( accountID );
+				expect( initialProperties ).toBeUndefined();
+
+				await untilResolved( registry, STORE_NAME ).getProperties( accountID );
+				expect( fetchMock ).toHaveFetched( propertiesEndpoint, { query: { accountID } } );
+
+				const properties = registry.select( STORE_NAME ).getProperties( accountID );
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+				expect( properties ).toEqual( fixtures.properties );
+				expect( properties ).toHaveLength( fixtures.properties.length );
+			} );
+
+			it( 'should not make a network request if properties for this account are already present', async () => {
+				const testAccountID = '12345';
+				const accountID = testAccountID;
+
+				// Load data into this store so there are matches for the data we're about to select,
+				// even though the selector hasn't fulfilled yet.
+				registry.dispatch( STORE_NAME ).receiveGetProperties( fixtures.properties, { accountID } );
+
+				const properties = registry.select( STORE_NAME ).getProperties( testAccountID );
+				await untilResolved( registry, STORE_NAME ).getProperties( testAccountID );
+
+				// It _may_ make a request for profiles internally if not loaded,
+				// so we only care that it did not fetch properties here.
```

Comment:
This seems to be copied from the analytics test (which is okay) except GA4 doesn't have profiles 😄 
```suggestion
```